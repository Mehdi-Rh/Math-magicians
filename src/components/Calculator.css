body {
  --grText: #61892f;
  --grBg: #86c232;
  --black: #222629;
  --darkGray: #474b4f;
  --lightGray: #6b6e70;
}

#calculator {
  display: inline-grid;
  width: 100%;
  margin: 5rem auto auto;
  font-size: xx-large;
  justify-content: center;
}

#screen {
  height: 5rem;
  background-color: var(--darkGray);
  color: white;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 0 1rem;
  font-size: 30px;
}

#keyboardContainer {
  display: -ms-grid;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.key {
  border: 0.1px solid var(--grText);
  color: var(--black);
  height: 5rem;
  width: 10rem;
  font-weight: 900;
  font-size: large;
}

.operation {
  background-color: var(--grBg);
  color: white;
  font-size: x-large;
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
  width: 20rem;
}

@media screen and (max-width: 768px) {
  .key {
    width: 20vw;
  }

  .zero {
    width: 40vw;
  }
}
