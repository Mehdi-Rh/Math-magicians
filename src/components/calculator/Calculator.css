body {
  --grText: #61892f;
  --grBg: #86c232;
  --black: #222629;
  --darkGray: #474b4f;
  --lightGray: #6b6e70;
}

#screen {
  height: 5rem;
  background-color: var(--darkGray);
  color: white;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 0 1rem;
  font-size: 30px;
}

#keyboardContainer {
  display: -ms-grid;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
}

.key {
  border: 0.1px solid var(--grText);
  color: var(--black);
  height: 5rem;
  width: 15vw;
  font-weight: 900;
  font-size: large;
}

.operation {
  background-color: var(--grBg);
  color: white;
  font-size: x-large;
}

.zero {
  grid-column-start: 1;
  grid-column-end: 3;
  width: 30vw;
}

@media screen and (min-width: 1000px) {
  .key {
    width: 10rem;
  }

  .zero {
    width: 20rem;
  }
}

@media screen and (max-width: 768px) {
  .key {
    width: 20vw;
  }

  .zero {
    width: 40vw;
  }
}
