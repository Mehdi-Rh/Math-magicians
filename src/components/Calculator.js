import React, { PureComponent } from 'react';
import './Calculator.css';

export default class Calculator extends PureComponent {
  render() {
    const keyboardValue = [
      ['AC', '+/-', '%', '/'],
      ['7', '8', '9', 'x'],
      ['4', '5', '6', '-'],
      ['1', '2', '3', '+'],
      ['0', '.', '='],
    ];

    const operationClass = (element, index) => (index === (element.length - 1) ? 'operation' : '');

    const zeroClass = (value) => (
      value === keyboardValue[4][0] ? 'zero' : ''
    );

    const keyboard = keyboardValue.map((row) => (

      row.map((key, keyIndex) => (
        <button type="submit" className={`key ${operationClass(row, keyIndex)} ${zeroClass(key)}`} key={key}>{key}</button>
      ))
    ));

    return (
      <div id="calculator">
        <div id="screen">0</div>
        <div id="keyboardContainer">
          { keyboard }
        </div>
      </div>
    );
  }
}
