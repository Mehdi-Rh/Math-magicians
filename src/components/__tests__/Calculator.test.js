import { fireEvent, render, screen } from '@testing-library/react';
import React from 'react';
import Renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('renders correctly', () => {
  const tree = Renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('shows 30 on "5 x 6 =" operation', () => {
  render(<Calculator />);

  fireEvent.click(screen.getByRole('button', { name: '5' }));
  fireEvent.click(screen.getByRole('button', { name: 'x' }));
  fireEvent.click(screen.getByRole('button', { name: '6' }));
  fireEvent.click(screen.getByRole('button', { name: '=' }));

  expect(screen.getByText('30')).toBeTruthy();
});
