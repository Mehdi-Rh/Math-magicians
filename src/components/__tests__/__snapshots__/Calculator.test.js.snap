// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="calculator"
>
  <div
    id="screen"
  >
    0
  </div>
  <div
    id="keyboardContainer"
  >
    <button
      className="key  "
      id="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="key  "
      id="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="key  "
      id="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="key operation "
      id="รท"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="key  "
      id={7}
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="key  "
      id={8}
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="key  "
      id={9}
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="key operation "
      id="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="key  "
      id={4}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="key  "
      id={5}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="key  "
      id={6}
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="key operation "
      id="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="key  "
      id={1}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="key  "
      id={2}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="key  "
      id={3}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="key operation "
      id="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="key  zero"
      id={0}
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="key  "
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="key operation "
      id="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
